src/apme_engine/engine/model_loader.py

```
@@ -1235,6 +1235,22 @@ def load_role(
+    if roleObj.metadata is None:
+        roleObj.metadata = {}
+    if isinstance(roleObj.metadata, dict) and not roleObj.metadata.get("argument_specs"):
+        for ext in ("yml", "yaml"):
+            arg_specs_path = os.path.join(fullpath, f"meta/argument_specs.{ext}")
+            if os.path.exists(arg_specs_path):
+                with open(arg_specs_path) as file:
+                    try:
+                        arg_specs_data = yaml.load(file, Loader=Loader)
+                        if isinstance(arg_specs_data, dict) and "argument_specs" in arg_specs_data:
+                            roleObj.metadata["argument_specs"] = arg_specs_data["argument_specs"]
+                    except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)